pkg/cmd/cli/delete_options.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/heptio/ark/pkg/client"
+	"github.com/heptio/ark/pkg/cmd/util/flag"
+	clientset "github.com/heptio/ark/pkg/generated/clientset/versioned"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+// DeleteOptions contains parameters used for deleting a restore.
+type DeleteOptions struct {
+	Names            []string
+	all              bool
+	Selector         flag.LabelSelector
+	Confirm          bool
+	Client           clientset.Interface
+	Namespace        string
+	singularTypeName string
+}
+
+func (o *DeleteOptions) NewDeleteOptions(singularTypeName string) {
```

Comment:
Let's remove the receiver from this function, and have it return a *DeleteOptions, so:
```
func NewDeleteOptions(singularTypeName string) *DeleteOptions {
    // instantiate and return a *DeleteOptions
}
```